core/src/main/java/org/lflang/util/FlexPRETUtil.java

```diff
@@ -0,0 +1,81 @@
+package org.lflang.util;
+
+import java.io.IOException;
+import java.util.Collections;
+import org.eclipse.xtext.xbase.lib.Exceptions;
+import org.lflang.FileConfig;
+import org.lflang.MessageReporter;
+import org.lflang.generator.GeneratorCommandFactory;
+import org.lflang.generator.LFGeneratorContext;
+import org.lflang.target.TargetConfig;
+import org.lflang.target.property.PlatformProperty;
+
+public class FlexPRETUtil {
+  private LFGeneratorContext context;
+  private GeneratorCommandFactory commandFactory;
+  private MessageReporter messageReporter;
+
+  public FlexPRETUtil(
+      LFGeneratorContext context,
+      GeneratorCommandFactory commandFactory,
+      MessageReporter messageReporter) {
+    this.context = context;
+    this.commandFactory = commandFactory;
+    this.messageReporter = messageReporter;
+  }
+
+  public void flashTarget(FileConfig fileConfig, TargetConfig config) {
+    var binPath = fileConfig.binPath;
+    var exeName = fileConfig.name;
+
+    /** FlexPRET's SDK generates a runnable bash script. We just need to run it. */
+    var cmd = binPath + "/" + exeName;
+    var args = Collections.<String>emptyList();
+
+    try {
+      LFCommand flash = commandFactory.createCommand(cmd, args);
+
+      /**
+       * Check if environment variable FP_SDK_FPGA_INTERFACE_PROGRAM is set. If it is set, we should
+       * let the user know it is not supported by LF.
+       *
+       * <p>We cannot support it because FP_SDK_FPGA_INTERFACE_PROGRAM will lead to a [Y/n] prompt,
+       * and the LFCommands do not have a way of getting the prompt response.
+       *
+       * <p>When FP_SDK_FPGA_INTERFACE_PROGRAM is not set, there is no prompt.
```

Comment:
This was a little unclear to me. If the user has set an environmental variable FP_SDK_FPGA_INTERFACE_PROGRAM then we cannot flash the executable. So the default behavior is that the variable is not set and we can flash? Under what circumstances will that variable be set? Manually be the user somehow?

Reply:
Hmm no, we can always flash the executable but we cannot support the [Y/n] prompt from this part of the `fpga-app.sh`: (because the LF command does not support stdin)

```
# Check if `xclip` exists
if ! command -v xclip > /dev/null 2 >& 1
then
    # Let the user know they can install `xclip` to automatically get command
    # copied to clipboard
    echo "Install \`xclip\` to automatically copy command to clipboard."
else
    # We ask to copy command; do not want to overwrite clipboard if user
    # has something important there
    echo "Copy command to clipboard? [Y/n]"
    read answer

    # Default response is yes
    if [ ! "$answer" != "${answer#[Nn]}" ] ; then 
        # Copy command to clipboard
        echo "$interface_cmd" | xclip -selection clipboard
        echo "Command copied to clipboard. Use CTRL + V to paste."
    fi
fi
```

The prompt only appears if the user has set the FP_SDK_FPGA_INTERFACE_PROGRAM environment variable. So in that case, we let the user know it is not supported.

Reply:
Why not just launch the FP_SDK_FPGA_INTERFACE_PROGRAM from the shell script instead of copying the launch command into the clipboard?

Reply:
Good question, hehe. I guess I considered doing this to be a little more versatile - with this approach the user can run commands between this script and actually running the "interface_cmd". But I realize now that use case is a bit narrow and probably not very useful. Maybe it's better to just run it automatically.

Reply:
I think this would be wise. What program (minicom/picocom/putty) that is used to write over the application is a detail that hardly matters for the user

Reply:
Fixed in https://github.com/pretis/flexpret/pull/96! :)